kobo/apps/openrosa/libs/utils/logger_tools.py

```diff
@@ -202,6 +209,13 @@ def create_instance(
     xml_hash = Instance.get_hash(xml)
     xform = get_xform_from_submission(xml, username, uuid)
     check_submission_permissions(request, xform)
+    if settings.STRIPE_ENABLED and check_usage_limits:
+        calculator = ServiceUsageCalculator(xform.user)
+        balances = calculator.get_usage_balances()
+        for usage_type in [UsageType.STORAGE_BYTES, UsageType.SUBMISSION]:
+            balance = balances[usage_type]
+            if balance and balance['exceeded']:
+                raise ExceededUsageLimitError()
```

Comment:
At which moment a balance does not exist? 

If it always exist, could it be simplified with something like

``` 
if any([
    balances[UsageType.STORAGE_BYTES]['exceeded'], 
    balances[UsageType. SUBMISSION]['exceeded'],
]):
    raise ExceededUsageLimitError()
```